templates/settings.py.erb

```diff
@@ -1,3 +1,16 @@
+################################################################################
+# !!! WARNING: DO NOT EDIT THIS FILE !!!                                       #
+################################################################################
+# File managed by Puppet.                                                      #
+# Module: pulpcore                                                             #
+################################################################################
+# Not only are your edits likely to be overwritten, there is also a strong     #
+# possibility of breaking your system if you change configuration here without #
+# making required changes elsewhere. Refer to the documentation you used to    #
+# install Pulpcore to determine the safe and persistent way to modify the      #
+# configuration.                                                               #
+################################################################################
```

Comment:
Is this part really needed? The user may not have explicitly thought about it so how are they going to link it back?

Reply:
I think it's important to provide some explanation to the user so that they understand why they are asked to not make edits.

If there is a warning not to make edits, but no explanation, a confident user might say "oh, I already understand it won't be persistent, so this warning doesn't apply to me. I just want to try out that other tasking system..."

I agree it's difficult to phrase it in such a way that is equally applicable to users of puppet, foreman-installer, satellite-installer, orcharhino, etc. Necessarily, the explanation will be vague, and this is the problem that I intend to solve by allowing custom strings in hiera.

But in the meantime, with the upcoming release introducing several new config file entries that can easily be broken in unexpected ways, I argue that a vague explanation is still better than no explanation at all.